src/Gui/ExpressionCompleter.cpp

```diff
@@ -541,7 +542,7 @@ class ExpressionCompleterModel: public QAbstractItemModel {
 
                 // if my element is a contextual descendant of root (current doc object list, current object prop list)
                 // mark it as such
-                if (element.row() >= docs.size()*2) {
+                if (element.row() >= static_cast<int>(docs.size()*2)) {
```

Comment:
Just for curiosity, how could you write
`)*2`
I am recently struggling with the formatter of Visual Studio. However the other rules we defined in FreeCAD's clang file are respected, in this case that there should be spaces around asterisks. Do you use especial formatting settings?
I am asking because I recently have some troubles with the formatter with unmatched braces and parentheses. I tried a lot to partially disable some formatting rules or to use another clang file. Therefore I am interested how you do this.

Reply:
I disabled clang-formatter in Visual Studio, as it was causing a lot of changes in the file that weren't needed. It was pushing mismatched '{' and '}', and I got tired of using CTRL-Z to undo the damage.

Reply:
> I disabled clang-formatter in Visual Studio, as it was causing a lot of changes in the file that weren't needed. It was pushing mismatched '{' and '}', and I got tired of using CTRL-Z to undo the damage.

that's is exactly the same issue I had. How did you disable the clang formatter while keeping the formatter running? I mean I need the formatter for indendations, to assure spaces round operators etc.
Last week I therefore disabled the automatic addion of braces. That fixed it for me but in general we want not to have braces also for single-line if statements and then automatic brace addition would help.

@chennes ,@Wandererfan